src/pybind/mgr/cephadm/module.py

```diff
@@ -634,20 +634,24 @@ def _run_cephadm(self, host, entity, command, args,
             if self.mode == 'root':
                 self.log.debug('args: %s' % (' '.join(final_args)))
                 self.log.debug('stdin: %s' % stdin)
-                script = 'injected_argv = ' + json.dumps(final_args) + '\n'
-                if stdin:
-                    script += 'injected_stdin = ' + json.dumps(stdin) + '\n'
-                script += self._cephadm
+                #script = b''
+                #if stdin:
+                #    script += b'injected_stdin = ' + json.dumps(stdin).encode('utf-8') + b'\n'
+                script = self._cephadm
                 python = connr.choose_python()
                 if not python:
                     raise RuntimeError(
                         'unable to find python on %s (tried %s in %s)' % (
                             host, remotes.PYTHONS, remotes.PATH))
                 try:
+                    # https://stackoverflow.com/questions/20276105
+                    loader = 'from tempfile import NamedTemporaryFile as T; import sys; t = T(); ' \
+                             't.write(sys.stdin.buffer.read()); t.flush(); sys.path.insert(0, t.name); ' \
+                             'import cephadm; cephadm.main(); t.close()'
```

Comment:
sigh.

Reply:
If this makes things even more complication, I'm very much inclined to re-think the current 'single-file' approach alltogether.. I guess it's time to take a cut now to make our lives easier in the long run. 

Going with a zipped files that gets distributed to all hosts on mgr startup is probably not the worst idea after all. We may even get a slight increase in launch time since we don't have to copy the sourcecode over _all_ the time.